src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1,394 +1,625 @@
-import static com.autotune.analyzer.recommendations.RecommendationConstants.RecommendationEngine.PercentileConstants.COST_MEMORY_PERCENTILE;
 import static com.autotune.analyzer.recommendations.RecommendationConstants.RecommendationValueConstants.*;
-import static com.autotune.analyzer.utils.AnalyzerConstants.PercentileConstants.*;
-
-public class CostRecommendationEngine implements KruizeRecommendationEngine {
-    private static final Logger LOGGER = LoggerFactory.getLogger(CostRecommendationEngine.class);
-    private String name;
-    private String key;
-    private RecommendationConstants.RecommendationCategory category;
-
-    public CostRecommendationEngine() {
-        this.name = RecommendationConstants.RecommendationEngine.EngineNames.COST;
-        this.key = RecommendationConstants.RecommendationEngine.EngineKeys.COST_KEY;
-        this.category = RecommendationConstants.RecommendationCategory.COST;
+import static com.autotune.analyzer.recommendations.RecommendationConstants.RecommendationValueConstants.MEM_ZERO;
+import static com.autotune.analyzer.utils.AnalyzerErrorConstants.AutotuneObjectErrors.MISSING_EXPERIMENT_NAME;
+
+public class RecommendationEngine {
+    private String performanceProfile;
+    private String experimentName;
+    private Timestamp intervalEndTime;
+    private Map<String, Terms> terms;
+    List<RecommendationModel> recommendationModels;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecommendationEngine.class);
+
+
+    public RecommendationEngine(String performanceProfile, String experimentName, Timestamp intervalEndTime, Map<String, Terms> terms) {
+        this.performanceProfile = performanceProfile;
+        this.experimentName = experimentName;
+        this.intervalEndTime = intervalEndTime;
+        this.terms = terms;
+        this.init();
     }
 
-    public CostRecommendationEngine(String name) {
-        this.name = name;
+    public RecommendationEngine() {
+
     }
 
-    /**
-     * Calculate the number of pods being used as per the latest results
-     * <p>
-     * pods are calculated independently based on both the CPU and Memory usage results.
-     * The max of both is then returned
-     *
-     * @param filteredResultsMap
-     * @return
-     */
-    private static int getNumPods(Map<Timestamp, IntervalResults> filteredResultsMap) {
-        Double max_pods_cpu = filteredResultsMap.values()
-                .stream()
-                .map(e -> {
-                    Optional<MetricResults> cpuUsageResults = Optional.ofNullable(e.getMetricResultsMap().get(AnalyzerConstants.MetricName.cpuUsage));
-                    double cpuUsageSum = cpuUsageResults.map(m -> m.getAggregationInfoResult().getSum()).orElse(0.0);
-                    double cpuUsageAvg = cpuUsageResults.map(m -> m.getAggregationInfoResult().getAvg()).orElse(0.0);
-                    double numPods = 0;
+    private void init() {
```

Comment:
init() 
inside init() methode call validation 
and if validation is sucess
Create KruizeObject 
and derive other parameters like
        datasource , metricProfile , Terms etc

Reply:
Done